src/shared-components/dropdown/index.tsx

```diff
@@ -34,6 +34,10 @@ interface DropdownProps<T> {
    * The handler to be invoked on option change
    */
   onChange: (option: T) => void;
+  /**
+   * The handler to be invoked on focus
+   */
+  onFocus?: (event?: React.FocusEvent) => void;
```

Comment:
Would it be worthwhile to rename this `onDropdownContainerFocus`? I might expect `onFocus` to cover focusing the individual dropdown element, but this is specifically for the container, so it might be helpful to distinguish it.